core/src/commonMain/kotlin/com/sourcepoint/mobile_core/Coordinator.kt

```diff
@@ -38,4 +61,255 @@ class Coordinator(
         repository.cachedMetaData = metaDataResponse.toString()
         return message
     }
+
+    private fun handleGetChoiceAll(response: ChoiceAllResponse, campaign: SPCampaignType) {
+        if (response.gdpr != null && campaign == SPCampaignType.Gdpr) {
```

Comment:
@Nevazhnovu giving it a second thought, I believe we don't need to check for 
```swift
response.gdpr != null && campaign == SPCampaignType.Gdpr
```
Just `response.gdpr` (and the other legislations) should be fine. This way we also don't need to receive `campaign` as a param.